internal/core/application/service.go

```diff
@@ -529,17 +529,21 @@ func (s *service) SubmitOffchainTx(
 				WithMetadata(errors.VtxoMetadata{VtxoOutpoint: vtxo.Outpoint.String()})
 		}
 
-		// check if already spent by another offchain tx
-		isSpent, err := s.cache.OffchainTxs().Includes(ctx, vtxo.Outpoint)
+		// check if already spent by another pending offchain tx
+		pendingTxid, err := s.cache.OffchainTxs().GetTxidByOutpoint(ctx, vtxo.Outpoint)
 		if err != nil {
 			log.WithError(err).Errorf("failed to check spent status of inputs against tx in cache")
 			return nil, errors.INTERNAL_ERROR.New("something went wrong").
 				WithMetadata(map[string]any{"vtxos": vtxo.Outpoint.String()})
 		}
 
-		if isSpent {
-			return nil, errors.VTXO_ALREADY_SPENT.New("%s already spent", vtxo.Outpoint.String()).
-				WithMetadata(errors.VtxoMetadata{VtxoOutpoint: vtxo.Outpoint.String()})
+		if pendingTxid != "" {
+			return nil, errors.VTXO_PENDING_TX.New(
+				"%s is locked by pending tx %s", vtxo.Outpoint.String(), pendingTxid,
```

Comment:
Let's be more generic, this way the error can be read by humans as it is, if not handled client-side

```suggestion
				"input(s) spent by another pending tx"
```